src/analysis/split.jl

```diff
@@ -52,15 +52,39 @@ function split_function_head(ex::Expr; source = nothing)
             name, args, kw, whereparams, _ = split_function_head(call)
             (name, args, kw, whereparams, rettype)
         end
-        Expr(:(::), arg::Symbol, argtype) || Expr(:(::), argtype) => (nothing, Any[ex], nothing, nothing, nothing)
         Expr(:where, call, whereparams...) => begin
             name, args, kw, _, rettype = split_function_head(call)
             (name, args, kw, whereparams, rettype)
         end
         _ => throw(SyntaxError("expect a function head, got $ex", source))
     end
 end
-split_function_head(s::Symbol; source = nothing) = (nothing, Any[s], nothing, nothing, nothing)
+split_function_head(s::Symbol; source=nothing) = (nothing, Any[s], nothing, nothing, nothing)
```

Comment:
can we also add some unit tests to cover these missing lines? not sure why this wasn't missing tho